pkg/handlers/webhookrhobsreceiver.go

```diff
@@ -0,0 +1,222 @@
+		return
+	}
+
+	metrics.ResetMetric(metrics.MetricRequestFailure)
+}
+
+func (h *WebhookRHOBSReceiverHandler) processAMReceiver(d AMReceiverData, ctx context.Context) *AMReceiverResponse {
+	log.WithField("AMReceiverData", fmt.Sprintf("%+v", d)).Info("Process alert data")
+
+	// Handle each firing alert
+	for _, alert := range d.Alerts.Firing() {
+		// Can we find a notification template for this alert?
+		templateName := alert.Labels[AMLabelTemplateName]
+		mfn := oav1alpha1.ManagedFleetNotification{}
+		err := h.c.Get(ctx, client.ObjectKey{
+			Namespace: OCMAgentNamespaceName,
+			Name:      templateName,
+		}, &mfn)
+		if err != nil {
+			log.WithError(err).Error("unable to locate corresponding notification template")
+			return &AMReceiverResponse{Error: err,
+				Status: fmt.Sprintf("unable to find ManagedFleetNotification %s", templateName),
+				Code:   http.StatusInternalServerError}
+		}
+
+		// Filter actionable alert based on Label
+		if !isValidAlert(alert, true) {
+			log.WithField(LogFieldAlert, fmt.Sprintf("%+v", alert)).Info("alert does not meet valid criteria")
+			continue
+		}
+
+		err = h.processAlert(alert, mfn)
+		if err != nil {
+			log.WithError(err).Error("a firing alert could not be successfully processed")
+		}
+	}
+	return &AMReceiverResponse{Error: nil, Status: "ok", Code: http.StatusOK}
+}
+
+// processAlert handles the pre-check verification and sending of a notification for a particular alert
+// and returns an error if that process completed successfully or false otherwise
+func (h *WebhookRHOBSReceiverHandler) processAlert(alert template.Alert, mfn oav1alpha1.ManagedFleetNotification) error {
+	fn := mfn.Spec.FleetNotification
+	mcID := alert.Labels[AMLabelAlertMCID]
+	hcID := alert.Labels[AMLabelAlertHCID]
+	slSent := false
+	for !slSent {
+
+		// Fetch the ManagedFleetNotificationRecord, or create it if it does not already exist
+		mfnr := &oav1alpha1.ManagedFleetNotificationRecord{}
+		err := h.c.Get(context.Background(), client.ObjectKey{
+			Namespace: OCMAgentNamespaceName,
+			Name:      mcID,
+		}, mfnr)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				log.WithError(err).Error("unable to fetch managedFleetNotificationRecord")
+				return fmt.Errorf("unable to fetch managedFleetNotificationRecord for %s", mcID)
+			}
+			// create ManagedFleetNotificationRecord if not found
+			mfnr, err = h.createManagedFleetNotificationRecord(mcID)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Fetch notificationRecordByName and ADD if it doesn't exist
+		nfr, err := mfnr.GetNotificationRecordByName(mcID, fn.Name)
+		if err != nil {
+			//  add NotificationRecordByName
+			nfr, err = addNotificationRecordByName(fn.Name, fn.ResendWait, hcID, mfnr)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Check if we already have a notification record for this hosted cluster
+		_, err = mfnr.GetNotificationRecordItem(mcID, fn.Name, hcID)
+		if err != nil {
+			// We mustn't, so create one
```

Comment:
small nit, what does the `We mustn't` here mean?

Reply:
Comments here have been updated to be clearer.